### PR TITLE
add: a warning message when using xformers in a PT 2.0 env.

### PR DESCRIPTION
**Pull request by sayakpaul**
See https://github.com/huggingface/diffusers/pull/3312#issuecomment-1536648693.

In addition to this, I think it makes sense to add PT 2.0 variant of attention processors for the following classes:

* `SlicedAttnProcessor`
* `SlicedAttnAddedKVProcessor`
* `LoRAAttnProcessor`
* `CustomDiffusionAttnProcessor`

Follow classes already have their PT 2.0 counterparts:

* `AttnProcessor` => `AttnProcessor2_0`
* `AttnAddedKVProcessor` => `AttnAddedKVProcessor2_0`

I can take care of it but wanted to check if @williamberman wants to do this as part of his attention refactor.

### CODE REVIEW

File: src/diffusers/models/attention_processor.py [hunk 1]
```diff
@@ -176,6 +178,11 @@ def set_use_memory_efficient_attention_xformers(
                     "torch.cuda.is_available() should be True but is False. xformers' memory efficient attention is"
                     " only available for GPU "
                 )
+            elif hasattr(F, "scaled_dot_product_attention") and self.scale_qk:
+                warnings.warn(
+                    "You have specified using flash attention using xFormers but you have PyTorch 2.0 already installed. "
+                    "So, we will default to native efficient flash attention implementation provided by PyTorch 2.0."
```
Comment (patrickvonplaten) [inline src/diffusers/models/attention_processor.py:184]:
```suggestion
                    "We will default to PyTorch's native efficient flash attention implementation provided by PyTorch 2.0."
```

**Review by patrickvonplaten — APPROVED** [review]
Thanks!

File: src/diffusers/models/attention_processor.py [hunk 2]
```diff
@@ -72,7 +73,8 @@ def __init__(
         self.upcast_attention = upcast_attention
         self.upcast_softmax = upcast_softmax
 
-        self.scale = dim_head**-0.5 if scale_qk else 1.0
+        self.scale_qk = scale_qk
+        self.scale = dim_head**-0.5 if self.scale_qk else 1.0
```
Comment (pcuenca) [inline src/diffusers/models/attention_processor.py:77]: Why this change? (Just curious)

**Review by pcuenca — APPROVED** [review]

File: src/diffusers/models/attention_processor.py (hunk 2 above)
Comment (sayakpaul) [reply src/diffusers/models/attention_processor.py:77]: Cannot use the `scale_qk` parameter in the latter methods otherwise. This parameter is for now needed to set the default attention processor to `AttnProcessor2_0`. 

I followed how we are setting `processor` in the init for that.

**Comment by sayakpaul** [comment]
@patrickvonplaten @pcuenca thanks!

WDYT about the following? 



> See [#3312 (comment)](https://github.com/huggingface/diffusers/pull/3312#issuecomment-1536648693).
> 
> In addition to this, I think it makes sense to add PT 2.0 variant of attention processors for the following classes:
> 
> * `SlicedAttnProcessor`
> * `SlicedAttnAddedKVProcessor`
> * `LoRAAttnProcessor`
> * `CustomDiffusionAttnProcessor`
> 
> Follow classes already have their PT 2.0 counterparts:
> 
> * `AttnProcessor` => `AttnProcessor2_0`
> * `AttnAddedKVProcessor` => `AttnAddedKVProcessor2_0`
> 
> I can take care of it but wanted to check if @williamberman wants to do this as part of his attention refactor.

Also, cc: @williamberman
